parcels/particle.py

```diff
@@ -457,6 +473,94 @@ def show(self, **kwargs):
+        # filename to saving to
+        savefile = kwargs.get('savefile', None)
+
+        if isinstance(t, datetime):
+            t = (t - self.grid.U.time_origin).total_seconds()
+        if isinstance(t, delta):
+            t = t.total_seconds()
+
+        if not np.isnan(latN):
+            latN = nearest_index(self.grid.U.lat, latN)
+            latS = nearest_index(self.grid.U.lat, latS)
+            lonE = nearest_index(self.grid.U.lon, lonE)
+            lonW = nearest_index(self.grid.U.lon, lonW)
+            lon = self.grid.U.lon[lonW:lonE]
+            lat = self.grid.U.lat[latS:latN]
+        else:
+            lon = self.grid.U.lon
+            lat = self.grid.U.lat
+
+        # time interpolation of velocity field
+        idx = self.grid.U.time_index(t)
+        U = np.array(self.grid.U.temporal_interpolate_fullfield(idx, t))
+        V = np.array(self.grid.V.temporal_interpolate_fullfield(idx, t))
+        if not np.isnan(latN):
+            U = U[latS:latN, lonW:lonE]
+            V = V[latS:latN, lonW:lonE]
+
+        # configuring plot
+        lat_median = np.median(lat)
+        lon_median = np.median(lon)
+        plt.figure()
+        m = Basemap(projection='merc', lat_0=lat_median, lon_0=lon_median,
+                    resolution='h', area_thresh=100,
+                    llcrnrlon=lon[0], llcrnrlat=lat[0],
+                    urcrnrlon=lon[-1], urcrnrlat=lat[-1])
+        if land:
+            m.drawcoastlines()
+            m.fillcontinents(color='burlywood')
+        parallels = np.arange(lat[0], lat[-1], abs(lat[0]-lat[-1])/5)
+        parallels = np.around(parallels, 2)
+        m.drawparallels(parallels, labels=[1, 0, 0, 0])
+        meridians = np.arange(lon[0], lon[-1], abs(lon[0]-lon[-1])/5)
+        meridians = np.around(meridians, 2)
+        m.drawmeridians(meridians, labels=[0, 0, 0, 1])
+
+        # formating velocity data for quiver plotting
+        U = np.array([U[y, x] for x in range(len(lon)) for y in range(len(lat))])
+        V = np.array([V[y, x] for x in range(len(lon)) for y in range(len(lat))])
+        speed = np.sqrt(U**2 + V**2)
+        normU = U/speed
+        normV = V/speed
+        x = np.repeat(lon, len(lat))
+        y = np.tile(lat, len(lon))
+
+        # plotting velocity vector field
+        vecs = m.quiver(x, y, normU, normV, speed, cmap=plt.cm.gist_ncar, clim=[0, vmax], scale=50, latlon=True)
+        cb = m.colorbar(vecs, "right", size="5%", pad="2%")
+        plon = [p.lon for p in self]
+        plat = [p.lat for p in self]
+        xs, ys = m(plon, plat)
+        # plotting particle data
+        m.scatter(xs, ys, color='black')
+
+        if(self.grid.U.time_origin == 0):
+            plt.title(delta(seconds=t))
+        else:
+            plt.title(netCDF4.num2date(t, 'seconds since '+str(self.grid.U.time_origin)))
+
+        if savefile is None:
+            plt.show()
+        else:
+            plt.savefig(savefile)
+            plt.close()
```

Comment:
Again, saving is a nice feature that would be useful more generally than just in `pset.show_velocity()` :-)